rotkehlchen/history/events/structures/asset_movement.py

```diff
@@ -0,0 +1,200 @@
+import logging
+from typing import TYPE_CHECKING, Any
+
+from rotkehlchen.accounting.mixins.event import AccountingEventType
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.constants import ZERO
+from rotkehlchen.constants.location_details import get_formatted_location_name
+from rotkehlchen.exchanges.data_structures import hash_id
+from rotkehlchen.history.events.structures.base import (
+    HISTORY_EVENT_DB_TUPLE_WRITE,
+    HistoryBaseEntry,
+    HistoryBaseEntryType,
+)
+from rotkehlchen.history.events.structures.types import (
+    HistoryEventSubType,
+    HistoryEventType,
+)
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_fval
+from rotkehlchen.types import (
+    Location,
+    TimestampMS,
+)
+from rotkehlchen.utils.misc import ts_ms_to_sec
+
+if TYPE_CHECKING:
+    from more_itertools import peekable
+
+    from rotkehlchen.accounting.mixins.event import AccountingEventMixin
+    from rotkehlchen.accounting.pot import AccountingPot
+    from rotkehlchen.fval import FVal
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class AssetMovement(HistoryBaseEntry):
+    """Asset movement event representing deposits and withdrawals on exchanges."""
+
+    def __init__(
+            self,
+            sequence_index: int,
+            timestamp: TimestampMS,
+            location: Location,
+            event_type: HistoryEventType,
+            event_subtype: HistoryEventSubType,
+            asset: Asset,
+            balance: Balance,
+            location_label: str | None = None,
```

Comment:
what is this used for here? If you don't know then you don't need to provide it as argument. One thing we used this for was the particular exchange name. So kraken1, coinbase2 etc. since user can have multiple api keys for exchange. If so then use that name in the argument.